src/api/cow/cow.spec.ts

```diff
@@ -0,0 +1,441 @@
+    }),
+    { status: HTTP_STATUS_NOT_FOUND }
+  )
+  try {
+    await cowSdk.cowApi.getPriceQuoteLegacy({
+      baseToken: 'unexistingToken',
+      quoteToken: 'unexistingToken',
+      amount: '1234567890',
+      kind: OrderKind.BUY,
+    } as PriceQuoteParams)
+  } catch (e) {
+    const error = e as QuoteError
+    expect(error.message).toEqual('Token pair selected has insufficient liquidity')
+    expect(fetchMock).toHaveBeenCalledTimes(1)
+    expect(fetchMock).toHaveBeenCalledWith(
+      'https://api.cow.fi/rinkeby/api/v1/markets/unexistingToken-unexistingToken/buy/1234567890',
+      {
+        body: undefined,
+        headers: {
+          'Content-Type': 'application/json',
+          'X-AppId': '0x0000000000000000000000000000000000000000000000000000000000000000',
+        },
+        method: 'GET',
+      }
+    )
+  }
+})
+
+test('Valid: Get Profile Data', async () => {
+  fetchMock.mockResponseOnce(JSON.stringify(PROFILE_DATA_RESPONSE), { status: HTTP_STATUS_OK })
+  const cowSdk1 = new CowSdk(chainId)
+  cowSdk1.updateChainId(1)
+
+  await cowSdk1.cowApi.getProfileData('0x6810e776880c02933d47db1b9fc05908e5386b96')
+  expect(fetchMock).toHaveBeenCalledTimes(1)
+  expect(fetchMock).toHaveBeenCalledWith(
+    'https://api.cow.fi/affiliate/api/v1/profile/0x6810e776880c02933d47db1b9fc05908e5386b96',
+    {
+      body: undefined,
+      headers: {
+        'Content-Type': 'application/json',
+        'X-AppId': '0x0000000000000000000000000000000000000000000000000000000000000000',
+      },
+      method: 'GET',
+    }
+  )
+})
+
+test('Invalid: Get Profile Data from unexisting address', async () => {
+  fetchMock.mockResponseOnce(
+    JSON.stringify({
+      errorType: 'NotFound',
+      description: "You've passed an invalid URL",
+    }),
+    { status: HTTP_STATUS_NOT_FOUND }
+  )
+  const cowSdk1 = new CowSdk(chainId)
+  cowSdk1.updateChainId(1)
+
+  try {
+    await cowSdk1.cowApi.getProfileData('unexistingAddress')
+  } catch (e) {
+    const error = e as CowError
+    expect(error.message).toEqual("You've passed an invalid URL")
+    expect(fetchMock).toHaveBeenCalledTimes(1)
+    expect(fetchMock).toHaveBeenCalledWith('https://api.cow.fi/affiliate/api/v1/profile/unexistingAddress', {
+      body: undefined,
+      headers: {
+        'Content-Type': 'application/json',
+        'X-AppId': '0x0000000000000000000000000000000000000000000000000000000000000000',
+      },
+      method: 'GET',
+    })
+  }
+})
+
+test('Invalid: Get Profile Data from not supported network', async () => {
+  const profileData = await cowSdk.cowApi.getProfileData('0x6810e776880c02933d47db1b9fc05908e5386b96')
+  expect(profileData).toBeNull()
+})
+
+test('Valid: Sign Order', async () => {
+  gpV2Contracts.signOrder.mockImplementation(() => ({
+    scheme: SigningScheme.EIP712,
+    data: '0x4d306ce7c770d22005bcfc00223f8d9aaa04e8a20099cc986cb9ccf60c7e876b777ceafb1e03f359ebc6d3dc84245d111a3df584212b5679cb5f9e6717b69b031b',
+  }))
```

Comment:
I'm not 100% convinced the signature should be mocked.
My argument against it is that the signature is handled internally and it doesn't need any external calls, thus, no need for mocking.
On the other hand, we could say that this is asserting that the API methods behave as expected, and doesn't care about the internal implementation of its individual methods, which should be tested separately.

What do you guys say? @ramirotw @matextrem

Reply:
maybe it's easier to setup a ethers.Wallet instance and have the `gpV2Contracts.signOrder` method to actually sign the order than to mock everything, since the sdk also works on nodejs all the local calls could be executed